exec/java-exec/src/main/java/org/apache/drill/exec/planner/sql/handlers/FindLimit0Visitor.java

```diff
@@ -46,6 +51,32 @@ public static boolean containsLimit0(RelNode rel) {
     return visitor.isContains();
   }
 
+  public static DrillRel addLimitOnTopOfLeafNodes(final DrillRel rel) {
+    final RelShuttleImpl shuttle = new RelShuttleImpl() {
+
+      private RelNode addLimitAsParent(RelNode node) {
+        final RexBuilder builder = node.getCluster().getRexBuilder();
+        final RexLiteral offset = builder.makeExactLiteral(BigDecimal.ZERO);
+        final RexLiteral fetch = builder.makeExactLiteral(BigDecimal.ZERO);
+        return new DrillLimitRel(node.getCluster(), node.getTraitSet(), node, offset, fetch);
```

Comment:
Agree with @jinfengni. In more recent versions of Calcite, use RelBuilder.limit() or .sortLimit(). The RelBuilder will be configured to create the appropriate Drill variants of all RelNodes. It might also do some useful canonization/optimization. We recommend using RelBuilder for most tasks involving creating RelNodes.